tests/app/main/test_suppliers.py

```diff
@@ -960,20 +953,20 @@ def test_filled_in_question_field_has_a_change_or_correct_a_mistake_link(
         page_html = response.get_data(as_text=True)
         document = html.fromstring(page_html)
         answer_required_link = document.xpath(
-            "//span[text()='{}']/following::td[2]/span/a[text()='Edit']".format(question)
+            f"//dt[normalize-space(text())='{question}']/following-sibling::dd[2]/a"
         )
 
         assert answer_required_link
-        assert answer_required_link[0].values()[0] == link_address
+        assert answer_required_link[0].values()[1] == link_address
 
     @pytest.mark.parametrize(
-        "summary,is_hint_visible",
+        "summary,link_text",
         [
-            ({"description": "Our company is the best for digital ponies."}, False),
-            ({"description": ""}, True),
+            ({"description": "Our company is the best for digital ponies."}, 'Change'),
+            ({"description": ""}, 'Change (optional)'),
         ]
     )
-    def test_hint_text_for_summary_only_visible_if_field_empty(self, summary, is_hint_visible):
+    def test_hint_text_for_summary_only_visible_if_field_empty(self, summary, link_text):
```

Comment:
we have replaced this 'hint text' with updated link text, so the test is updated accordingly.